state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2147,8 +2156,91 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		origin: DepositOrigin<T, I>,
 	) -> DepositAction<T, I> {
 		match action.clone() {
-			ChannelAction::LiquidityProvision { lp_account, .. } => {
-				T::Balance::credit_account(&lp_account, asset.into(), amount_after_fees.into());
+			ChannelAction::LiquidityProvision { lp_account, additional_action, refund_address } => {
+				let used_for_flip_funding_swap = if let Some(AdditionalDepositAction::FundFlip {
+					flip_amount_to_credit,
+					role_to_register,
+				}) = additional_action
+				{
+					let is_flip_asset = matches!(asset.into(), Asset::Flip);
+					let funding_amount =
+						if is_flip_asset { amount_after_fees.into() } else { INITIAL_FLIP_FUNDING };
+
+					T::FundAccount::fund_account(
+						lp_account.clone(),
+						None,
+						funding_amount.into(),
+						FundingSource::InitialFunding,
+					);
+					if let Some(role) = role_to_register {
+						if let Err(err) =
+							T::AccountRoleRegistry::register_account_role(&lp_account, role)
+						{
+							log::warn!("Failed to register account role: {err:?}");
```

Comment:
This will print a warning in all the node logs, not ideal. I think we can just silently ignore this case, or we emit an event to notify the user.